src/cryptonote_core/blockchain.cpp

```diff
@@ -5349,6 +5433,12 @@ void Blockchain::set_user_options(uint64_t maxthreads, bool sync_on_blocks, uint
   m_max_prepare_blocks_threads = maxthreads;
 }
 
+void Blockchain::set_txpool_notify(TxpoolNotifyCallback&& notify)
+{
+  std::lock_guard<decltype(m_txpool_notifier_mutex)> lg(m_txpool_notifier_mutex);
```

Comment:
I used `std::lock_guard` to not further cement Boost dependencies, and since `std::mutex` and `std::lock_guard` are already used within the codebase, I think it shouldn't affect binary size. However, I'm not incredibly opinionated either way.

Reply:
We're already using it all over the place.